browser.js

```diff
@@ -10,22 +10,47 @@ const urls = {
 	v6: 'https://ipv6.icanhazip.com/'
 };
 
+const backupUrls = {
+	v4: 'https://api.ipify.org',
+	v6: 'https://api6.ipify.org'
+};
+
 const queryHttps = (version, options) => {
 	let xhr;
 	const promise = new Promise((resolve, reject) => {
 		const doReject = () => {
 			reject(new Error('Couldn\'t find your IP'));
 		};
 
+		const tryBackup = () => {
+			xhr = new XMLHttpRequest();
+			xhr.addEventListener('error', doReject, {once: true});
+			xhr.addEventListener('timeout', doReject, {once: true});
+
+			xhr.addEventListener('load', () => {
+				const ip = xhr.responseText.trim();
+
+				if (!ip || !isIp[version](ip)) {
+					doReject();
+				}
+
+				resolve(ip);
+			}, {once: true});
+
+			xhr.open('GET', backupUrls[version]);
+			xhr.timeout = options.timeout;
+			xhr.send();
+		};
```

Comment:
Too much duplication. You could generalize the XHR stuff into a reusable async function that could be used for both the main and backup.